Loop/Managers/DeviceDataManager.swift

```diff
@@ -332,11 +332,15 @@ final class DeviceDataManager {
             .store(in: &cancellables)
         
         // Turn off preMeal when going into closed loop off mode
+        // Cancel any active temp basal when going into closed loop off mode
         // The dispatch is necessary in case this is coming from a didSet already on the settings struct.
         $isClosedLoop
             .removeDuplicates()
             .receive(on: DispatchQueue.main)
-            .sink { if !$0 { self.loopManager.settings.clearOverride(matching: .preMeal) } }
+            .sink { if !$0 {
+                self.loopManager.settings.clearOverride(matching: .preMeal)
+                self.loopManager.cancelActiveTempBasal()
```

Comment:
Just a couple ideas about organizing responsibilities. 1) isClosedLoop might be more of a LoopAppManager level responsibility, now that we have that split out.  2) Dosing decisions are LDM's responsibility. LDM could be told "closed loop is off", and it can decide to "cancel active temp basal", if that is the appropriate action.

I think the current setup works, so I'll approve.

Reply:
I like it and tried to implement. Let me know what you think.